webpack/components/AnsibleHostDetail/components/AnsibleVariableOverrides/AnsibleVariableOverridesHelper.js

```diff
@@ -0,0 +1,8 @@
+export const reorderVariables = variables =>
+  variables.reduce((memo, role) => {
+    const vars = role?.ansibleVariablesWithOverrides?.nodes;
+    if (vars) {
+      return memo.concat(vars);
+    }
+    return memo;
+  }, []);
```

Comment:
We could use `flatMap` here which doesn't create an intermediate array for each iteration (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#alternative) and is more readable:
```suggestion
roles.flatMap((role)=>(role.ansibleVariablesWithOverrides.nodes))
```
If I'm not mistaken, the conditional chaining operand is not needed since GraphQL guarantees that the response will always include these fields even if they are empty.

Reply:
Updated, conditional chaining is not needed as mentioned in [previous comment](https://github.com/theforeman/foreman_ansible/pull/420#discussion_r665386867)